### PR TITLE
Some macro to function conversions

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: RichString.h [hunk 1]
```diff
@@ -15,13 +15,13 @@ in the source distribution for its full text.
 #define RichString_size(this) ((this)->chlen)
 #define RichString_sizeVal(this) ((this).chlen)
 
-#define RichString_begin(this) RichString (this); memset(&this, 0, sizeof(RichString)); (this).chptr = (this).chstr;
-#define RichString_beginAllocated(this) memset(&this, 0, sizeof(RichString)); (this).chptr = (this).chstr;
+#define RichString_begin(this) RichString (this); memset(&(this), 0, sizeof(RichString)); (this).chptr = (this).chstr;
```
Comment (BenBE) [inline RichString.h:18]:
```suggestion
#define RichString_begin(this) RichString (this); RichString_beginAllocated((this));
```

File: RichString.h [hunk 2]
```diff
@@ -15,13 +15,13 @@ in the source distribution for its full text.
...
+#define RichString_beginAllocated(this) memset(&(this), 0, sizeof(RichString)); (this).chptr = (this).chstr;
 #define RichString_end(this) RichString_prune(&(this));
 
 #ifdef HAVE_LIBNCURSESW
 #define RichString_printVal(this, y, x) mvadd_wchstr(y, x, (this).chptr)
-#define RichString_printoffnVal(this, y, x, off, n) mvadd_wchnstr(y, x, (this).chptr + off, n)
+#define RichString_printoffnVal(this, y, x, off, n) mvadd_wchnstr(y, x, (this).chptr + (off), n)
```
Comment (BenBE) [inline RichString.h:24]:
```suggestion
#define RichString_printoffnVal(this, y, x, off, n) mvadd_wchnstr((y), (x), (this).chptr + (off), (n))
```

File: linux/LinuxProcessList.c [hunk 3]
```diff
@@ -1172,7 +1172,7 @@ static inline void LinuxProcessList_scanZfsArcstats(LinuxProcessList* lpl) {
    char buffer[128];
    while (fgets(buffer, 128, file)) {
       #define tryRead(label, variable) do { if (String_startsWith(buffer, label) && sscanf(buffer + strlen(label), " %*2u %32llu", variable)) { break; } } while(0)
-      #define tryReadFlag(label, variable, flag) do { if (String_startsWith(buffer, label) && sscanf(buffer + strlen(label), " %*2u %32llu", variable)) { flag = 1; break; } else { flag = 0; } } while(0)
+      #define tryReadFlag(label, variable, flag) do { if (String_startsWith(buffer, label) && sscanf(buffer + strlen(label), " %*2u %32llu", variable)) { (flag) = 1; break; } else { (flag) = 0; } } while(0)
```
Comment (BenBE) [inline linux/LinuxProcessList.c:1175]:
```suggestion
      #define tryReadFlag(label, variable, flag) do { if (String_startsWith((buffer), (label)) && sscanf((buffer) + strlen((label)), " %*2u %32llu", (variable))) { (flag) = 1; break; } else { (flag) = 0; } } while(0)
```

File: RichString.h (hunk 1 above)
Comment (cgzones) [reply RichString.h]: added, thanks

File: RichString.h (hunk 2 above)
Comment (cgzones) [reply RichString.h:24]: from my understanding it is not necessary to enclose these function arguments, as comma has the lowest operator preference

File: linux/LinuxProcessList.c (hunk 3 above)
Comment (cgzones) [reply linux/LinuxProcessList.c:1175]: `buffer` is not a marco argument (it comes from the callers scope) and i think it is unnecessary to enclose single function arguments

Comment (BenBE) [reply linux/LinuxProcessList.c:1175]: k, for `buffer`. But `label` and `variable` are macro arguments and thus should be enclosed.

File: RichString.h (hunk 2 above)
Comment (BenBE) [reply RichString.h:24]: Mostly suggested to avoid the "syntax injection" issue with macros in general.

**Comment by cgzones** [comment]
Travis succeeded at https://travis-ci.com/github/cgzones/htop/builds/194819344

**Pull request merged**